### PR TITLE
Add pollers for `CopyAndWait` in `file/files` and `blobs/blob`

### PR DESCRIPTION
**Pull request by catriona-m**

### CODE REVIEW

File: storage/2020-08-04/blob/blobs/copy_and_wait_poller.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	getPropertiesInput GetPropertiesInput
+}
+
+func (p *copyAndWaitPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	props, err := p.client.GetProperties(ctx, p.containerName, p.blobName, p.getPropertiesInput)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving properties (container: %s blob: %s) : %+v", p.containerName, p.blobName, err)
+	}
+
+	if strings.EqualFold(string(props.CopyStatus), "success") {
```
Comment (tombuildsstuff) [inline storage/2020-08-04/blob/blobs/copy_and_wait_poller.go:36]: could we use the constant here:

```suggestion
	if strings.EqualFold(string(props.CopyStatus), string(Success)) {
```

File: storage/2020-08-04/file/files/copy_and_wait_poller.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	fileName  string
+}
+
+func (p *copyAndWaitPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	props, err := p.client.GetProperties(ctx, p.shareName, p.path, p.fileName)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving copy (shareName: %s path: %s fileName: %s) : %+v", p.shareName, p.path, p.fileName, err)
+	}
+
+	if strings.EqualFold(props.CopyStatus, "success") {
```
Comment (tombuildsstuff) [inline storage/2020-08-04/file/files/copy_and_wait_poller.go:36]: (same here) can we use the constant:

```suggestion
	if strings.EqualFold(props.CopyStatus, string(Success)) {
```

File: storage/2023-11-03/blob/blobs/copy_and_wait_poller.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	getPropertiesInput GetPropertiesInput
+}
+
+func (p *copyAndWaitPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	props, err := p.client.GetProperties(ctx, p.containerName, p.blobName, p.getPropertiesInput)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving properties (container: %s blob: %s) : %+v", p.containerName, p.blobName, err)
+	}
+
+	if strings.EqualFold(string(props.CopyStatus), "success") {
```
Comment (tombuildsstuff) [inline storage/2023-11-03/blob/blobs/copy_and_wait_poller.go:36]: (same here) can we use the Constant value?

```suggestion
	if strings.EqualFold(string(props.CopyStatus), string(Success)) {
```

File: storage/2023-11-03/file/files/copy_and_wait_poller.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	fileName  string
+}
+
+func (p *copyAndWaitPoller) Poll(ctx context.Context) (*pollers.PollResult, error) {
+	props, err := p.client.GetProperties(ctx, p.shareName, p.path, p.fileName)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving copy (shareName: %s path: %s fileName: %s) : %+v", p.shareName, p.path, p.fileName, err)
+	}
+
+	if strings.EqualFold(props.CopyStatus, "success") {
```
Comment (tombuildsstuff) [inline storage/2023-11-03/file/files/copy_and_wait_poller.go:36]: (same here) can we use the Constant value?

```suggestion
	if strings.EqualFold(string(props.CopyStatus), string(Success)) {
```

**Review by tombuildsstuff — APPROVED** [review]
A few minor point about using the constant value, but this otherwise LGTM 👍

File: storage/2020-08-04/file/files/copy_and_wait_poller.go (hunk 2 above)
Comment (catriona-m) [reply storage/2020-08-04/file/files/copy_and_wait_poller.go:36]: we don't have a constant for this in files atm, just blobs,  is it worth adding for this?

Comment (tombuildsstuff) [reply storage/2020-08-04/file/files/copy_and_wait_poller.go:36]: Oh ok, I figured this was still around/used - in which case this is fine 👍

**Pull request merged**